server/src-lib/Hasura/GraphQL/Schema/Action.hs

```diff
@@ -337,43 +336,47 @@ mkActionsSchema =
   (\aggregate actionInfo ->
      case _adType $ _aiDefinition actionInfo of
        ActionQuery ->
-         Map.foldrWithKey (accumulateQuery (_aiPgScalars actionInfo)) aggregate $ mkQueryActionSchemaOne actionInfo
+         Map.foldrWithKey (accumulateQuery (_aiPgScalars actionInfo)) aggregate $
+           mkQueryActionSchemaOne actionInfo
        ActionMutation kind ->
-         Map.foldrWithKey (accumulateMutation (_aiPgScalars actionInfo)) aggregate $ mkMutationActionSchemaOne actionInfo kind
+         Map.foldrWithKey (accumulateMutation (_aiPgScalars actionInfo)) aggregate $
+           mkMutationActionSchemaOne actionInfo kind
   )
   mempty
   where
     -- we'll need to add uuid and timestamptz for actions
-    mkNewRoleState pgScalars =
+    initRoleState =
       ( mempty
-      , foldr addScalarToTyAgg mempty $
-        pgScalars <> Set.fromList [PGJSON, PGTimeStampTZ, PGUUID]
+      , foldr addScalarToTyAgg mempty [PGJSON, PGTimeStampTZ, PGUUID]
       )
```

Comment:
This function is causing the issue. The `mkNewRoleState` function initiates the state with empty `RootFields` and `TyAgg` with PG scalars reused and mandatory scalars for actions.This function is invoked only for the first action in the list (unfortunately) and remaining actions' reused PG scalars were not added. Hence the issue only exist if multiple actions are defined. The fix is to move adding PG scalars here to the accumulate functions where the PG scalars are added to `TyAgg` explicitly for each action.